### PR TITLE
Added trigger logic information to PMT decoder diagnostic ROOT tree

### PR DESCRIPTION
**Pull request by PetrilloAtWork**
With the new adder-based trigger, the PMT diagnostic tree should know whether the light collected at beam gate was coming from an adder or a majority trigger.
This commit adds that information to the tree, one piece of information per cryostat, directly from the trigger information.

_Edit_: it also adds a count of triggers from the beginning of the run and the gates from the previous trigger. This tree is being used for trigger studies as well.
_And_ a job configuration to create such trees, for the benefit of grid jobs.

Reviewer: @mvicenzi as PMT decoder insider.

### CODE REVIEW

File: icaruscode/Decode/DaqDecoderICARUSPMT_module.cc [hunk 1]
```diff
@@ -3340,7 +3331,6 @@ void icarus::DaqDecoderICARUSPMT::initFragmentsTree() {
   tree->Branch("waveformSize", &data.waveformSize);
   tree->Branch("triggerBits", &data.triggerBits);
   tree->Branch("triggerSource", &data.triggerSource);
-  tree->Branch("triggerLocation", &data.triggerLocation);
```
Comment (mvicenzi) [inline icaruscode/Decode/DaqDecoderICARUSPMT_module.cc:3343]: Ehm... it seems you are removing that information, instead of adding it?

**Review by mvicenzi — CHANGES_REQUESTED** [review]
I'm confused... it seems the updated code was already in develop and now you are removing it?

**Comment by PetrilloAtWork** [comment]
:flushed:
There was clearly an issue with my final branching (noted the three PR in a row coming from the same working area?). In fact, I pretty much lost the source of this PR, except that luckily I had tarballed it for grid jobs.

I have updated it now... thank you for catching this!!

There may or may not be a following commit rebasing this same code to the current `develop` branch.

File: fcl/reco/Stage0/Run2/partial/decodePMT_icarus_treeonly.fcl [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+# In the plain ROOT output file (names `Tree-....root`) a diagnostic tree
+# is stored with one entry per PMT data fragment (see `decodePMT_icarus.fcl`
+# and the decoder module `DaqDecodeICARUSPMT`), as `daqPMT/PMTfragments`.
+#
+# All standard trigger and PMT decoding data products are produced with the
+# standard names (`daqTrigger` and `daqPMT` respectively).
+# The noticeable exception is PMT raw waveforms not the corrections:
+# neither are saved.
+#
+# The log file also dump the trigger information.
```
Comment (mvicenzi) [inline fcl/reco/Stage0/Run2/partial/decodePMT_icarus_treeonly.fcl:34]:
```suggestion
# The log file also dumps the trigger information.
```

File: fcl/reco/Stage0/Run2/partial/decodePMT_icarus_treeonly.fcl [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+# Output
+# -------
+# 
+# In the plain ROOT output file (names `Tree-....root`) a diagnostic tree
+# is stored with one entry per PMT data fragment (see `decodePMT_icarus.fcl`
+# and the decoder module `DaqDecodeICARUSPMT`), as `daqPMT/PMTfragments`.
+#
+# All standard trigger and PMT decoding data products are produced with the
+# standard names (`daqTrigger` and `daqPMT` respectively).
+# The noticeable exception is PMT raw waveforms not the corrections:
```
Comment (mvicenzi) [inline fcl/reco/Stage0/Run2/partial/decodePMT_icarus_treeonly.fcl:31]:
```suggestion
# The noticeable exception is PMT raw waveforms and the corrections:
```

**Review by mvicenzi — APPROVED** [review]
Way better now!

**Comment by mmrosenberg** [comment]
trigger build

**Comment by mmrosenberg** [comment]
trigger build

**Comment by mmrosenberg** [comment]
trigger build

**Pull request merged**